### PR TITLE
Eigen: update to 3.3.9

### PR DESCRIPTION
**Pull request by NeroBurner**
ceres-solver 1.14.0 fails to build with Eigen 3.3.8. See link for more info

https://gitlab.com/libeigen/eigen/-/issues/2011

The fix is part of Eigen 3.3.9

* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **[Yes]**

Disabled toolchains:
- `nmake-vs-15-2017-win64-cxx17`
---

### CODE REVIEW

**Comment by NeroBurner** [comment]
vs-15-2017-win64-cxx17 is the only pipeline failing because of a missing fortran compiler

https://github.com/cpp-pm/hunter/pull/323/checks?check_run_id=1556327497

I've NO idea why Eigen needs (or checks) for a fortran compiler

**Comment by NeroBurner** [comment]
all green with disabled `nmake-vs-15-2017-win64-cxx17` toolchain

@rbsheth please review/merge

**Comment by NeroBurner** [comment]
Just tested Eigen v3.3.8 builds and it shows, that the toolchain already failed with 3.3.8

- PR: https://github.com/cpp-pm/hunter/pull/324
- failing `nmake-vs-15-2017-win64-cxx17` build: https://github.com/cpp-pm/hunter/pull/324/checks?check_run_id=1557649819

File: cmake/projects/Eigen/ci/matrix.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[
```
Comment (rbsheth) [inline cmake/projects/Eigen/ci/matrix.json:1]: It is okay for a toolchain to fail. You should remove this file.

**Review by rbsheth — COMMENTED** [review]
@NeroBurner As you can see here https://cpp-pm.github.io/hunter/ the nmake toolchain was broken. You don't need a PR like #324 for testing that. :)

Please remove matrix.json here - we want to log failures on the above page.

**Comment by NeroBurner** [comment]
uuuuuh, fancy! Totally forgot about that page!

Removed the commit with the removed toolchain

File: cmake/projects/Eigen/ci/matrix.json (hunk 1 above)
Comment (NeroBurner) [reply cmake/projects/Eigen/ci/matrix.json]: removed it

**Review by rbsheth — APPROVED** [review]

**Comment by rbsheth** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.288

**Pull request merged**